### ISSUE TITLE
Backport request to incorporate SQL Server Deadlock fix #1448 in spring-batch 4.3.x / spring-boot-starter-batch 2.6.x

### ISSUE THREAD

**Issue by darkmastermindz** [spring-projects/spring-batch#4085]
See: #1448 and ref: @benas in #3927 

**Context**

How has this issue affected you?
Unable to use spring batch for enterprise project for compatibility reasons as organization currently supports up to JDK11 and stable versions in compatibility.

What are you trying to accomplish?
Incorporate fix with spring batch that runs on JDK11 or lower / GA version as a minor bug fix.

What other alternatives have you considered?
- Replacing Spring Batch all-together
- Overwriting GA version classes

Are you aware of any workarounds?
No

**Comment by benas** [comment]
Thank you for this request. However, the fix in #1448 is a breaking change and we cannot backport it in a patch release of the v4 line.

What you can do is create the sequences and register the new incrementer (which is based on sequences) through the job repository factory bean, something like:

```
@Bean
public BatchConfigurer batchConfigurer(DataSource dataSource) {
	return new DefaultBatchConfigurer(dataSource) {
		@Override
		protected JobRepository createJobRepository() throws Exception {
			JobRepositoryFactoryBean factoryBean = new JobRepositoryFactoryBean();
			factoryBean.setIncrementerFactory(new DefaultDataFieldMaxValueIncrementerFactory(dataSource) {
				@Override
				public DataFieldMaxValueIncrementer getIncrementer(String incrementerType, String incrementerName) {
					return new SqlServerSequenceMaxValueIncrementer(dataSource, incrementerName);
				}
			});
			// set other properties on the factory bean
			factoryBean.afterPropertiesSet();
			return factoryBean.getObject();
		}
	};
}
```

This should work in a similar way to what is described here: https://github.com/spring-projects/spring-framework/issues/21425#issuecomment-453472597.

I'm closing this issue for now, but do not hesitate to add a comment if you need further support on this.

**Comment by karan-s9** [comment]
FYI in case it helps anyone.  In addition to what @benas mentioned above, you also need to override JobLauncher and set the job repository to the overridden bean that uses SqlServerSequenceMaxValueIncrementer. If you only override JobRepository, spring-batch will assume BATCH_JOB_SEQ and BATCH_JOB_EXECUTION_SEQ are still tables instead of sequences.
